Add a test for `bswap`

adpaco-aws opened the pull request:
### Description of changes: 

Adds a test for `bswap`, which relies on CBMC's implementation. As in other bit-manipulation intrinsics, it suffers from #934 

### Resolved issues:

Part of #727 

### Call-outs:

<!-- 
Address any potentially confusing code. Is there code added that needs to be cleaned up later? Is there code that is missing because it’s still in development? 
-->

### Testing:

* How is this change tested? Adds one test.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

celinval commented on tests/kani/Intrinsics/bswap.rs:
```diff
@@ -0,0 +1,45 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Check that `bswap` returns the expected results
```
Can you please state here what that result is? `bswap` is not the most intuitive name. It could be bit / byte swap. I had to google to check exactly what this does.

celinval commented on tests/kani/Intrinsics/bswap.rs:
```diff
@@ -0,0 +1,45 @@
+}
+
+#[kani::proof]
+fn main() {
```
Can you split these into multiple harnesses again?

celinval commented on tests/kani/Intrinsics/bswap.rs:
```diff
@@ -0,0 +1,45 @@
+                bytes_res
+        }
+
+        // Checks that the original byte is equal to the swapped byte
```
Maybe change this to say checks that the order of the bytes in the original integer is reverse in the swapped one.

celinval commented on tests/kani/Intrinsics/bswap.rs:
```diff
@@ -0,0 +1,45 @@
+        // Checks that the original byte is equal to the swapped byte
+        fn $check_name(a: $ty, b: $ty) {
+            let bytes = std::mem::size_of::<$ty>();
+            let i: usize = kani::any();
```
I like how you used any here. Much cleaner than an iterator of some sort.

adpaco-aws replied:
And more efficient too!

adpaco-aws commented:
Extended the description, added an example and link to the intrinsic documentation. Also split into several proofs (one per macro call), so all comments should be addressed now.

celinval reviewed (approved):
Thanks!
